trieste/models/gpflow/architectures.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" This file contains builders for some GPflow models supported in Trieste. """
+
+from __future__ import annotations
+
+from typing import Optional
+
+import gpflow
+import tensorflow as tf
+import tensorflow_probability as tfp
+from gpflow.models import GPR, SGPR, SVGP, VGP
+
+from ...data import Dataset
+from ...space import Box
+
+
+def build_gpr(data: Dataset, trainable_likelihood: bool = False, kernel_priors: bool = True) -> GPR:
```

Comment:
1. trainable_likelihood: if we keep a single Boolean argument, it should be interpolate = T/F. If True, then variance=1e-5 (or less) and trainable=False. Otherwise it's the unknown noise scenario and we just train the likelihood variance. There is also a scenario where noise variance is known. We can do that if we use 2 arguments: trainable_likelihood=T/F, likelihood_variance= optional float. We could then recover the interpolating mode by setting trainable=F, variance=0. I would say the choice depends on the usage we have in the notebooks?
2. priors: do we want to make those optional? more generally, do we want to have lots of possible choices for the user, or just a few scenarii? If we go for more flexibility, we might also add the kernel as an optional input.

Reply:
I think having a single reasonable prior is okay

Reply:
Would this also be a good place to allow data scaling?

Reply:
I think I would go with simpler build functions here, with some options but not too much (hence single prior and single boolean argument)  - since the aim here was mostly to give novice users good gpflow models, but perhaps this is something to discuss?

say this is for us to use, what would we need in terms of flexibility? if its actually not that much (say few more kernels and prior types) perhaps we can add a bit more flexibility

Reply:
@henrymoss good point about scaling, that occurred to me as well since I'm reviewing the data transform PR at the moment, not sure what to think about it yet